### ISSUE TITLE
Read of netCDF file fails with units attribute that is not of type string

### ISSUE THREAD

**Issue by ajwdewit** [pydata/xarray#4634]
When opening a netCDF file which has a variable that has a units attribute with a definition of integer 1, xarray fails to open the file:

```
~/bin/miniconda3/envs/py37_crucial/lib/python3.8/site-packages/xarray/conventions.py in _update_bounds_attributes(variables)
    379     for v in variables.values():
    380         attrs = v.attrs
--> 381         has_date_units = "units" in attrs and "since" in attrs["units"]
    382         if has_date_units and "bounds" in attrs:
    383             if attrs["bounds"] in variables:

TypeError: argument of type 'numpy.int64' is not iterable
```

This because line 381 will search for the word "since" in an integer type which fails. Of course assigning a integer as units attribute is not wise, but some people tend to assign integer 1 for variables which have no units.

**Comment by mathause** [comment]
Yes, that could probably be updated to 

```diff
- has_date_units = "units" in attrs and "since" in attrs["units"]
+ has_date_units = "units" in attrs and and isinstance(attrs["units"], str) and "since" in attrs["units"]
```

are you interested to send in a PR?

**Comment by ajwdewit** [comment]
I have an NC file lying around so I can easily test it. I will send a PR in coming days.

**Comment by DocOtak** [comment]
Keep in mind that the [NetCDF user guide](https://www.unidata.ucar.edu/software/netcdf/docs/attribute_conventions.html) "strongly recommends" that units be a character string. The [CF Conventions](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.8/cf-conventions.html#attribute-appendix) requires the units to be a character string. I think in xarray you can set `decode_cf=False` or `decode_times=False` in the various `open_*` methods to turn off this interpretation if you need things to work right now.

**Comment by dcherian** [comment]
Hmmm... but see http://cfconventions.org/Data/cf-conventions/cf-conventions-1.8/cf-conventions.html#units

```
 The conforming unit for quantities that represent fractions, or parts of a whole, is "1". 
```

**Comment by DocOtak** [comment]
@dcherian Often I find it a little easier to understand the [Conformance Document](http://cfconventions.org/Data/cf-documents/requirements-recommendations/conformance-1.8.html#units), bullet point two says:
>The type of the units attribute is a string that must be recognizable by the udunits package. Exceptions are the units level, layer, and sigma_level.

This shouldn't prevent xarray from doing something useful with non conforming files if it can.

**Comment by kmuehlbauer** [comment]
Should be fixed by #7085. Please reopen, if this is still an issue.
